fivox/eventSource.h

```diff
@@ -158,6 +159,21 @@ class EventSource
      */
     float getDt() const;
 
+    /** @return the current time from setTime() in milliseconds. */
+    float getCurrentTime() const;
+
+    /**
+     * Load and update events for the given chunks of the data source.
+     *
+     * @param chunkIndex the chunk to start loading from
+     * @param numChunks the number of chunks to load into memory
+     * @return the number of updated events, or -1 if the load failed.
+     */
+    ssize_t load( size_t chunkIndex, size_t numChunks );
```

Comment:
Why don't you make this function virtual and remove _load if the implementation is just calling _load? I don't see the point of the current design.